### ISSUE TITLE
topgrade should first upgrade itself and relaunch

### ISSUE THREAD

**Issue by drrlvn** [r-darwish/topgrade#32]

**Comment by r-darwish** [comment]
This needs some discussion before we can implement it.

There are multiple ways to obtain Topgrade:

* Using your system package manager - currently only available on Arch, but I do plan to add an Homebrew recipe and will be happy if someone makes a PPA for Ubuntu
* Using `cargo install` - either from Github or crates.io
* Downloading a precompiled binary and putting it somewhere in your PATH

In each of these there's a different way to upgrade Topgrade. The first question is whether Topgrade can detect which method was used to install it without making mistakes.

The second question is what should Topgrade do for each method.

My first idea was to split Topgrade into two phases. The first one invokes the system package manager and cargo upgrade and then always relaunches, but I'll need some way to pass my summary information from the first phase to the second phase. Also, if a new feature was introduced to the "first phase", the auto-upgrade won't catch it.

**Comment by r-darwish** [comment]
I think this is no longer needed. You can now install topgrade from Brew, AUR and `cargo install`. In all cases, updates will be handled by Topgrade itself. I can probably make a PPA/COPR/Port/EBUILD/whatever if necessary.

While you can use the precompiled binaries, that's not really the recommended installation method, so I think it's redundant to implement self-update just for the case that you're using the precompiled binaries.
